### PR TITLE
[PAPP-32695] Jira: Bugfix -Adding support for Jira error messages stored in HTTP Response object

### PR DESCRIPTION
**Pull request by bbielinski-splunk**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [x] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Adding support for Jira error messages stored in HTTP Response object [PAPP-32695]

## What is the current behavior? (OPTIONAL)
n/a

## What is the new behavior? (OPTIONAL)
n/a

## Other information (OPTIONAL)
n/a

## Pay close attention to (OPTIONAL)
n/a

## Screenshots (if relevant)
n/a
---
Thanks for contributing!

### CODE REVIEW

File: jira_view.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # either express or implied. See the License for the specific language governing permissions
 # and limitations under the License.
 def get_ctx_result(provides, result):
-    """ Function that parses data.
+    """Function that parses data.
```
Comment (mposluszny-splunk) [inline jira_view.py:16]: Can we make this descriptions more descriptive? Probably many functions in programming world parse some data

File: jira_connector.py [hunk 2]
```diff
@@ -362,8 +393,8 @@ def _set_jira_error(self, result_object, message, e):
 
     def _create_jira_object(self, action_result):
 
-        if (not self._verify_cert):
-            if ('REQUESTS_CA_BUNDLE' in os.environ):
+        if not self._verify_cert:
+            if 'REQUESTS_CA_BUNDLE' in os.environ:
```
Comment (mposluszny-splunk) [inline jira_connector.py:397]: let's merge those ifs

File: jira_connector.py [hunk 3]
```diff
@@ -331,28 +331,59 @@ def _save_state(self, state):
...
 
     def _set_jira_error(self, result_object, message, e):
-
-        error_text = None
-
-        try:
+        # Newer versions of Jira API may not include the error messages in
+        # e.text. Instead, these messages are included in e.response,
+        # which is a Response object from requests package.
+        if hasattr(e, "text") and e.text:
```
Comment (mposluszny-splunk) [inline jira_connector.py:337]: I'd split error text retrieval to two separate function.

File: jira_connector.py (hunk 2 above)
Comment (bbielinski-splunk) [reply jira_connector.py:397]: Done

File: jira_view.py (hunk 1 above)
Comment (bbielinski-splunk) [reply jira_view.py:16]: Done

File: jira_connector.py (hunk 3 above)
Comment (bbielinski-splunk) [reply jira_connector.py:337]: Done

File: jira_view.py [hunk 4]
```diff
@@ -25,45 +26,46 @@ def get_ctx_result(provides, result):
...
+    ctx_result["param"] = param
 
     if summary:
-        ctx_result['summary'] = summary
+        ctx_result["summary"] = summary
 
-    ctx_result['action'] = provides
+    ctx_result["action"] = provides
 
     if not data:
```
Comment (mposluszny-splunk) [inline jira_view.py:36]: We can simplify 36 - 39 as 
`ctx_result["data"] = data if data else []`

Comment (bbielinski-splunk) [reply jira_view.py:36]: Good catch, applied

**Review by mposluszny-splunk — APPROVED** [review]

**Review by tonyc-phantom — APPROVED** [review]

**Pull request merged**